framework/test/unit/specs/modules/keys/keys_module.spec.ts

```diff
@@ -11,3 +11,523 @@
  *
  * Removal or modification of this copyright notice is prohibited.
  */
+import { LiskValidationError } from '@liskhq/lisk-validator';
+import { codec } from '@liskhq/lisk-codec';
+import { getRandomBytes, hash } from '@liskhq/lisk-cryptography';
+import { Account, GenesisBlock, Transaction, transactionSchema } from '@liskhq/lisk-chain';
+import { objects as ObjectUtils } from '@liskhq/lisk-utils';
+import { KeysModule } from '../../../../../src/modules/keys/keys_module';
+import { createFakeDefaultAccount, StateStoreMock } from '../../../../utils/node';
+import * as fixtures from './fixtures.json';
+import { GenesisConfig } from '../../../../../src';
+import { genesisBlock as createGenesisBlock } from '../../../../fixtures/blocks';
+import { AccountKeyAsset } from '../../../../../src/modules/keys/types';
+
+// eslint-disable-next-line import/order
+import cloneDeep = require('lodash.clonedeep');
+
+describe('keys module', () => {
+	let decodedMultiSignature: any;
+	let validTestTransaction: any;
+	let targetMultisigAccount: any;
+	let stateStore: any;
+	let keysModule: KeysModule;
+	let reducerHandler: any;
+	let decodedBaseTransaction: any;
+
+	const defualtTestCase = fixtures.testCases[0];
+
+	const genesisConfig: GenesisConfig = {
+		baseFees: [
+			{
+				assetType: 0,
+				baseFee: '1',
+				moduleType: 3,
+			},
+		],
+		bftThreshold: 67,
+		blockTime: 10,
+		communityIdentifier: 'lisk',
+		maxPayloadLength: 15360,
+		minFeePerByte: 1,
+		rewards: {
+			distance: 1,
+			milestones: ['milestone'],
+			offset: 2,
+		},
+	};
+
+	beforeEach(() => {
+		keysModule = new KeysModule(genesisConfig);
+		const buffer = Buffer.from(defualtTestCase.output.transaction, 'base64');
+		const id = hash(buffer);
+		decodedBaseTransaction = codec.decode<Transaction>(transactionSchema, buffer);
+		decodedMultiSignature = {
+			...decodedBaseTransaction,
+			id,
+		};
+		validTestTransaction = new Transaction(decodedMultiSignature);
+
+		targetMultisigAccount = createFakeDefaultAccount({
+			address: Buffer.from(defualtTestCase.input.account.address, 'base64'),
+			balance: BigInt('94378900000'),
+		});
+
+		stateStore = new StateStoreMock();
+
+		stateStore.account = {
+			get: jest.fn().mockResolvedValue(targetMultisigAccount),
+			getOrDefault: jest.fn().mockResolvedValue(
+				createFakeDefaultAccount({
+					address: Buffer.from(defualtTestCase.input.account.address, 'base64'),
+				}) as never,
+			),
+		};
+
+		reducerHandler = {};
+	});
+
+	describe('beforeTransactionApply', () => {
+		it('should not fail to validate valid signatures', async () => {
```

Comment:
`beforeTransactionApply` has 3 main cases, it should be clear which cases it's testing.
and maybe missing some cases?

Reply:
Can we split into https://github.com/LiskHQ/lisk-sdk/issues/5653 and we groom it to have a complete list of all the cases (probably all the ones on this PR plus whatever was missed in the past)